libafl_sugar/src/forkserver.rs

```diff
@@ -177,25 +177,24 @@ impl<'a, const MAP_SIZE: usize> ForkserverBytesCoverageSugar<'a, MAP_SIZE> {
             // A fuzzer with feedbacks and a corpus scheduler
             let mut fuzzer = StdFuzzer::new(scheduler, feedback, objective);
 
+            let forkserver = if self.shmem_testcase {
```

Comment:
Maybe you can assign the builder to a variable, then use `shmem_provider().build()` or just `build()`, depending on the input
Or should there be an additonal flag to use or not use the provider?